concordium-consensus/tests/consensus/ConcordiumTests/LeaderElectionTest.hs

```diff
@@ -0,0 +1,147 @@
+                        predictLeadershipElectionNonce ss 7 (Just 11) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 7 Nothing 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 8) 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 11) 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 Nothing 15 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 8) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 (Just 11) 20 `shouldBe` Nothing,
+                        predictLeadershipElectionNonce ss 5 Nothing 20 `shouldBe` Nothing
+                    ]
+    where
+        ss = SeedState {
+            epochLength = 9,
+            epoch = 0,
+            currentLeadershipElectionNonce = H.hash "1",
+            updatedNonce = H.hash "2"
+        }
+        vrfKP = fst $ VRF.randomKeyPair (mkStdGen 0)
+        bn = VRF.prove vrfKP "3"
+        ss' = updateSeedState 10 bn ss
+        ss'' = updateSeedState 7 bn ss
+
+genSeedstate :: Gen SeedState
+genSeedstate =
+    do
+        (el :: Word64) <- (*9) <$> choose (1,10000)
+        epoch <- choose (0, 1000)
+        return SeedState {
+            epochLength = fromIntegral el,
+            currentLeadershipElectionNonce = H.hash "1",
+            updatedNonce = H.hash "2",
+            ..
+        }
+
+
+-- Tests that both Maybe List's are Just, that the second Just is a singleton,
+-- and that the element of the singleton is an element of the first list.
+singletonElementOfProperty :: Maybe [LeadershipElectionNonce] -> Maybe [LeadershipElectionNonce] -> Property
+singletonElementOfProperty (Just a) (Just [b]) = counterexample "Prediction from second seedstate should be one of the predictions in the first." $ elem b a
+singletonElementOfProperty _ _ = counterexample "Both arguments should be Just, and the second Just should contain a list with one element." False
+
+-- Generates a seedstate meant to be from a block in the last third of an epoch and tests @predictLeadershipElectionNonce@
+-- on this seedstate and a target slot in the next epoch. It tests against @computeLeadershipElectionNonce@ and tests that
+-- @predictLeadershipElectionNonce@ gives the same prediction on an updated seedstate for the same target slot.
+makePropertyJust :: Property
+makePropertyJust = property $ do
+    ss@SeedState{..} <- genSeedstate
+    let epochSlot = epoch * fromIntegral epochLength
+    let nextEpoch = epochSlot + fromIntegral epochLength
+    let toThirds = 2 * fromIntegral epochLength `div` 3
+    -- Find a slot in the last 2/3 of the epoch
+    seedStateSlot :: Slot <- fromIntegral <$> choose (epochSlot + toThirds, nextEpoch-2) -- We substract 2 so that we can update the seedstate and still be in the same epoch.
```

Comment:
```suggestion
    let twoThirds = 2 * fromIntegral epochLength `div` 3
    -- Find a slot in the last 2/3 of the epoch
    seedStateSlot :: Slot <- fromIntegral <$> choose (epochSlot + twoThirds, nextEpoch-2) -- We subtract 2 so that we can update the seedstate and still be in the same epoch.
```